assets/src/data/eventespresso/core/reducers/dirty-entities.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * External imports
+ */
+import { DEFAULT_CORE_STATE } from '@eventespresso/model';
+import { fromJS, Set } from 'immutable';
+import { normalizeEntityId } from '@eventespresso/helpers';
+
+/**
+ * Internal imports.
+ */
+import { ACTION_TYPES } from '../actions/action-types';
+const { entities: types } = ACTION_TYPES;
+
+/**
+ * Handle adding incoming data to state.
+ *
+ * @param {Immutable.Map} state
+ * @param {string} modelName
+ * @param {number} entityId
+ * @param {Immutable.Set} existingEntities
+ * @return {Immutable.Map} New state.
+ */
+const addToState = ( state, modelName, entityId, existingEntities ) => {
+	existingEntities = existingEntities.add( entityId );
+	return state.set( modelName, existingEntities );
+};
+
+/**
+ * Handle removing incoming data from state
+ *
+ * @param {Immutable.Map} state
+ * @param {string} modelName
+ * @param {number|string} entityId
+ * @return {Immutable.Map} new state or existing state if no change.
+ */
+const removeFromState = ( state, modelName, entityId ) => {
+	let entityIds = state.get( modelName, Set() );
+	if ( ! entityIds.includes( entityId ) ) {
+		return state;
```

Comment:
> ie: it is assumed that a previous call to removeFromState() with an existing entityId would have successfully removed the empty model after the entity was removed?

That.  We are bailing early if the entityId isn't present in the collection because it means there is nothing to do (and any previous removals would have removed the model from the state).